lib/Doctrine/Common/Collections/ArrayCollection.php

```diff
@@ -71,6 +71,17 @@ public function toArray()
         return $this->elements;
     }
 
+    /**
+     * @return list<T>
+     */
+    public function toList(): array
+    {
+        if (\PHP_VERSION_ID >= 80100 && \array_is_list($this->elements)) {
+            return $this->elements;
+        }
```

Comment:
Did you benchmark this code? Is `array_is_list()` really so much faster than `array_values()`? If not, I'd remove this block.